### PR TITLE
Feature: Listing card styling and general favorite icon (toggle favorited on click)

### PR DESCRIPTION
**Pull request by Alex-AK**
Add listing card component structure
Add listing card styling
Update column styling
Add star icon SVG for favorite state (dynamic active and inactive based on state).
Add PUT request to star icon on click (toggle favorited for listing).

NOTES: 
- This PR is reliant on the backend PR 'bug/fix-knexfile'
- When clicking the star icon the PUT request succeeds. However, the star will not update automatically -- the listing reducer needs to be updated to map the column data to update the listing in the store to update the UI on click.

### CODE REVIEW

File: src/modules/common/FavoritedIcon.jsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  listingId: number,
+  favorited: Boolean,
+};
+
+const FavoritedIcon = ({ listingId, favorited }: Props) => {
+  // dispatch
+  const dispatch = useDispatch();
+
+  // toggle favorited on listing
+  const updateFavorited = () => dispatch(editListing(listingId, { favorited: !favorited }));
```
Comment (nateboyette) [inline src/modules/common/FavoritedIcon.jsx]: Request to edit listing is failing. Will investigate.

Comment (Alex-AK) [reply src/modules/common/FavoritedIcon.jsx]: The request to update is succeeding, we need to update the reducer state map to properly update the UI. I wrote up a little blurb in the description of this PR. Unless there's a 500 coming back on your end I think we're good for this PR?

Comment (nateboyette) [reply src/modules/common/FavoritedIcon.jsx]: I was testing manually from dev before seeing the note. Merged the backend Pr's and all requests are working now.

**Comment by nateboyette** [comment]
Rebased on master. Fixed merge conflict

**Pull request merged**